pkg/cmd/pulumi/plugin/plugin_install.go

```diff
@@ -172,6 +193,31 @@ func (cmd *pluginInstallCmd) Run(ctx context.Context, args []string) error {
 				diag.Message("", "Plugin download URL set to %s"), pluginSpec.PluginDownloadURL)
 		}
 
+		if pluginSpec.Kind == apitype.ResourcePlugin && // The registry only supports plugin types
+			!cmd.env.GetBool(env.DisableRegistryResolve) &&
+			pluginSpec.PluginDownloadURL == "" { // Don't override explicit pluginDownloadURLs
+			pkgMetadata, err := registry.ResolvePackageFromName(ctx, cmd.registry, pluginSpec.Name, pluginSpec.Version)
+			if err == nil {
+				pluginSpec.Name = pkgMetadata.Name
+				pluginSpec.PluginDownloadURL = pkgMetadata.PluginDownloadURL
+			} else if errors.Is(err, registry.ErrNotFound) && !strings.ContainsRune(pluginSpec.Name, '/') {
+				// If the error is [registry.ErrNotFound], then this could
+				// be a Pulumi plugin that isn't in the registry.
+				//
+				// Ideally, we would have a hard-coded list such plugins
+				// so we could provide useful error messages when someone
+				// requests a non-existent plugin.
```

Comment:
Maybe a dynamic list in the registry that we could check against, but I really don't think we should have a hard-coded list in `pulumi/pulumi` for this.

Reply:
And we're sure this is *never* going to change?  Having a long hardcoded list somewhere feels a bit scary to me tbh.

Reply:
Yes, because the list exists only to ensure backward compatibility. It will prevent calls of `pulumi plugin install resource <name>` from breaking if they already worked. 

The list exists to capture _now_, so it shouldn't need to be updated for the future.

For a new provider from Pulumi, we will need to publish to the registry to enable `pulumi plugin install resource <name>`, but that isn't breaking.